controllers/dbaas.redhat.com/crunchybridgeconnection_controller.go

```diff
@@ -47,10 +54,60 @@ type CrunchyBridgeConnectionReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *CrunchyBridgeConnectionReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = log.FromContext(ctx)
+	logger := log.FromContext(ctx, "CrunchyBridgeConnection", req.NamespacedName)
 
-	// your logic here
+	var connection dbaasredhatcomv1alpha1.CrunchyBridgeConnection
+	err := r.Get(ctx, req.NamespacedName, &connection)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			logger.Info("CrunchyBridgeConnection resource not found, has been deleted")
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "Error fetching CrunchyBridgeConnection for reconcile")
+		return ctrl.Result{}, err
+	}
+	inventory := dbaasredhatcomv1alpha1.CrunchyBridgeInventory{}
+	if err := r.Get(ctx, types.NamespacedName{Namespace: req.Namespace, Name: connection.Spec.InventoryRef.Name}, &inventory); err != nil {
+		if apierrors.IsNotFound(err) {
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			logger.Info("inventory resource not found, has been deleted")
+			return ctrl.Result{}, err
+		}
+		logger.Error(err, "Error fetching CrunchyBridgeConnection for reconcile")
+		return ctrl.Result{}, err
+	}
+	instance := getInstance(&inventory, connection.Spec.InstanceID)
+	if instance == nil {
+		return ctrl.Result{}, err
```

Comment:
err has not been updated for this block, is this intentional or should it be emitting a newer, custom error?

Reply:
updated, now returning the error from getInstance.